### DISCUSSION TITLE
Is Theme-UI dying..?

### DISCUSSION THREAD

**Discussion by pkieltyka** [system-ui/theme-ui#1207] [General] [answered]
Hi there, I'm just wondering if theme-ui is still being maintained and in active development? I am surprised by the slow pace its taking to port to TS given that most of the work is already completed, but seems no urgency to actually make a release? Further, there are a growing number of issues and PRs that are also unattended..?

it would be helpful to understand the intentions and plan or I'll likely migrate away from theme-ui

**Comment by atanasster** [comment, marked as answer]
i am just a (very active) user, and it doesn' seem theme-ui is dying as it keeps getting (small) updates, including for TS. Special thanks to @hasparus . What parts of theme-ui are you using?

But I do agree it seems a bit disconnected.  I am not sure who is actually making "releases" and what are the steps to close up on a release.

For my part I would love to contribute in some way, but am rarely getting any feedback or direction where the lib is going or much of a sense of community. Some of the more successful OSS projects do have an advisory board of some sore and a roadmap, maybe this will help?

**Comment by deadcoder0904** [comment]
Would love to get an update on this from the team :)

**Comment by hasparus** [comment]
Okay, I think I can shed some light on the situation.

jxnblk has given me permissions to publish Theme UI to NPM few days ago (it was a bit harded than expected as we had a few hiccups with Lerna and NPM orgs), so we can start releasing bugfixes to 0.3 and deliver TS rewrite with 0.4.

I'm saying __we__, as I will certainly fail to do it alone, as I'm a bit preocuppied with protesting (:hankey: has hit the fan in the place I live), a full time job to pay the bills and other projects. Fortunately, I can think of a few talented prior contributors we could get in touch with. I hope we can push something forward together.

As of the question... Code once published to NPM is immutable, and will live forever in the package ~~heaven~~ registry.
I'm using Theme UI in a bunch of projects, and in some of them the migration to other styling solution like Stitches or Tailwind would be impossible.

> Some of the more successful OSS projects do have an advisory board of some sore and a roadmap, maybe this will help

This is certainly a good idea. 

> For my part I would love to contribute in some way, but am rarely getting any feedback or direction where the lib is going or much of a sense of community.

@atanasster, would you like to talk in DMs or by a Zoom/Whereby call to figure out how you could contribute?

**Reply by atanasster** [reply]
@hasparus - thanks a lot for the update - I am sure most OOS projects hit some roadblocks at some point in their existence, so it was mostly the lack of feedback that was worrisome. I am glad you have a lot invested in theme-ui already and plan on keeping using it, I am in a similar boat :)

Yes, I would like to get on a chat/zoom, I am in Florida on East coast timezone but a bit of a late-morning person.

**Comment by hasparus** [comment]
> it would be helpful to understand the intentions and plan or I'll likely migrate away from theme-ui

@pkieltyka, I suppose there could be two reasons for migrating away now

1. Performance. It would take a lot of effort to get Theme UI close to build-time solutions performance like `@compiled/css-in-js` or Tailwind. Precompiling `sx` prop in babel plugin is possible, but quite hard.
2. No support for JSX automatic runtime, as Emotion doesn't have that yet. It's an annoying pain point, because `automatic` is the new default.

**Reply by pkieltyka** [reply]
Emotion 11 is going to be shipping their final release soon and I believe has support for new JSX transform.

My issue with theme-ui is the development has stagnated very much and yet it’s so close to accomplishing the version port to typescript to offer a better experience.

I agree a compiled css with a similar experience to theme-ui would be awesome and maybe that’ll come in future works as underlining libraries mature.. another one to track is https://linaria.dev/

Finally I’ve recently been looking to switch away from theme-ui given that it looks like it’s hit a wall in dev, and came across chakra ui which is clearly heavily inspired by theme-ui but is much more actively maintained.

FYI, if there was a roadmap for theme-ui with a real effort, I’d be happy to sponsor/donate

**Reply by deadcoder0904** [reply]
Thanks, @hasparus for your contributions. Unfortunately, I switched to Tailwind 2 days ago as I hit a few roadblocks. I'll probably come back when it gets more matured :)

**Reply by hasparus** [reply]
@deadcoder0904 Thanks! Tailwind is great though. Frankly speaking, I'd recommend it over any CSS-in-JS library any time you don't _need_ your styles to be in JS / don't need them at runtime.

**Reply by dcastil** [reply]
Support for automatic JSX runtime is coming in https://github.com/system-ui/theme-ui/pull/1237. 😊

**Reply by atanasster** [reply]
@dcastil thats awesome, i was really looking forward to this.
